### PR TITLE
feat(builder): add toml file support

### PR DESCRIPTION
**Pull request by jkzing**
# PR Details

- add compiled dependency `toml-loader`
- add builder plugin toml which uses toml-loader to load toml files

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/webpack-builder/src/plugins/toml.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export const PluginToml = (): BuilderPlugin => ({
+  name: 'webpack-builder-plugin-toml',
+
+  setup(api) {
+    api.modifyWebpackChain((chain, { CHAIN_ID }) => {
+      chain.module
+        .rule(CHAIN_ID.RULE.TOML)
+        .test(/\.toml$/)
+        .use(CHAIN_ID.USE.TOML)
+        .loader(require.resolve('../../compiled/toml-loader'));
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/src/plugins/toml.ts:12]: 这里可以使用 modifyWebpackChain 传入的 getCompiledPath 工具函数哈

File: packages/toolkit/utils/src/chainId.ts [hunk 2]
```diff
@@ -25,6 +25,8 @@ export const CHAIN_ID = {
     SVG: 'svg',
     SVG_INLINE: 'svg-inline',
     SVG_URL: 'svg-url',
+    /** Rule for toml */
+    TOML: 'toml',
```
Comment (chenjiahan) [inline packages/toolkit/utils/src/chainId.ts:29]: 修改 modern-js 里的 package 后，需要在根目录执行 pnpm run change 来添加一下 changeset 中英文描述~
（webpack-builder 除外，目前还没正式发包，所以不需要添加）

File: packages/builder/webpack-builder/src/plugins/toml.ts (hunk 1 above)
Comment (jkzing) [reply packages/builder/webpack-builder/src/plugins/toml.ts:12]: get

Comment (jkzing) [reply packages/builder/webpack-builder/src/plugins/toml.ts]: 看起来好多地方都没用这个，得统一改一波 🥺

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
👏🏻👏🏻

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**